tests/playbooks/tests_route_device.yml

```diff
@@ -0,0 +1,188 @@
+              route"
+        - name: Get the interface1 MAC address
+          command: cat /sys/class/net/{{ interface1 }}/address
+          register: interface1_mac
+          ignore_errors: true  # noqa ignore-errors
+          changed_when: false
+        - name: Configure the IP addresses and the route with only the MAC
+            address specified
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface1 }}"
+                mac: "{{ interface1_mac.stdout }}"
+                type: ethernet
+                autoconnect: false
+                ip:
+                  address:
+                    - 198.51.100.4/24
+                    - 2001:db8::6/32
+                  route:
+                    - network: 198.58.10.64
+                      prefix: 26
+                      gateway: 198.51.100.102
+                      metric: 4
+
+        - name: Assert that the warning about specifying the route without
+            the output device is logged for initscripts provider
+          assert:
+            that:
+              - __network_connections_result.stderr is search("The connection
+                .* does not specify an interface name. Therefore, the route .*
+                will be configured without the output device and the kernel
+                will choose it automatically which might result in an unwanted
+                device being used. To avoid this, specify `interface_name` in
+                the connection appropriately.")
+            msg: The warning about specifying the route without the output
+              device is not logged for initscripts provider
+          when: network_provider == "initscripts"
+
+        - name: Assert that no warning is logged for nm provider
+          assert:
+            that:
+              - __network_connections_result.stderr is not search("<warn>")
+            msg: The warning is logged for nm provider
+          when: network_provider == "nm"
+      always:
+        - name: Remove test configuration
+          block:
+            - name: Bring down test devices and profiles
+              include_role:
+                name: linux-system-roles.network
+              vars:
+                network_connections:
+                  - name: "{{ interface0 }}"
+                    persistent_state: absent
+                    state: down
+                  - name: "{{ interface1 }}"
+                    persistent_state: absent
+                    state: down
+              ignore_errors: true  # noqa ignore-errors
+            - name: Assert interface0 profile and interface1 profile are absent
+              include_tasks: tasks/assert_profile_absent.yml
+              vars:
+                profile: "{{ item }}"
+              loop:
+                - "{{ interface0 }}"
+                - "{{ interface1 }}"
```

Comment:
When using ignore_errors above (which seems to be correct), it does not seem to be necessary to assert that the profiles are gone. And if, then maybe only after trying to delete everything to ensure that as much as possible is cleaned up.

Reply:
I honestly think that we should clean up all the tests interfaces and profiles at the end of the tests so that we know that the current tests will not affect the next running tests (imagine that we are running the test playbooks in sequences). Therefore, from my point of view, it is necessary to assert the profiles are absent.

Reply:
In that case, it does not seem to make sense to ignore the errors in line 166. Also, your response does not address the concern to first delete the interfaces before asserting that the profiles are gone.

Reply:
I dropped the ignore_errors in line 166. I also moved the assertion (about the profile absent) after deleting the interface.